scheduler/proxy/proxy.go

```diff
@@ -0,0 +1,49 @@
+package proxy
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"gocloud.dev/pubsub"
+)
+
+type MessageMutateFunc func(*pubsub.Message) (*pubsub.Message, error)
+
+type PubSubProxy struct {
+	topic        *pubsub.Topic
+	subscription *pubsub.Subscription
+}
+
+func New(topic *pubsub.Topic, subscription *pubsub.Subscription) *PubSubProxy {
+	return &PubSubProxy{
+		topic:        topic,
+		subscription: subscription,
+	}
+}
+
+func (proxy *PubSubProxy) Listen(ctx context.Context, preprocess MessageMutateFunc) error {
+	for {
+		msg, err := proxy.subscription.Receive(ctx)
+		if err != nil {
+			log.Println("error receiving message: ", err)
+			continue
```

Comment:
This is a problem that existed prior to your PR as well, but if Receive gets an error, it'll never succeed again per: https://pkg.go.dev/gocloud.dev/pubsub#Subscription.Receive

We need to either return here or recreate the subscription before retrying.

Reply:
I've returned here to avoid the application hanging with a broken Subscription, going forwards we should look to reconstruct the Subscription before retrying as you mention.